apps/src/music/views/Generate.tsx

```diff
@@ -53,29 +59,63 @@ const Generate: React.FunctionComponent = () => {
     packId
   );
 
-  const generateSong = useCallback(() => {
+  const generateSongAi = useCallback(async () => {
     console.log('starting ask');
-    dispatch(setAiGenerateState('generating'));
-    askAi(
+    const result = await askAi(
       `Here is the context:
   ${contextGenerateMusicPsuedocodeFromDescription}
 
   And here is the request:
   ${useText}`
-    ).then(result => {
-      if (result.length > 1 && result[1].status === AiInteractionStatus.OK) {
-        console.log(result[1].chatMessageText);
-        const psuedocode = result[1].chatMessageText.replaceAll('```', '');
-        const resultBlockly = generateBlocklyJson(psuedocode);
-        dispatch(setCodeToLoad(resultBlockly));
-      } else {
-        console.error('Error getting AI response.');
-      }
-
-      dispatch(setAiGenerateState('done'));
-    });
+    );
+    if (result.length > 1 && result[1].status === AiInteractionStatus.OK) {
+      console.log(result[1].chatMessageText);
+      const psuedocode = result[1].chatMessageText.replaceAll('```', '');
+      const resultBlockly = generateBlocklyJson(psuedocode);
+      dispatch(setCodeToLoad(resultBlockly));
```

Comment:
Minor: could factor these two `generate` functions into something that just returns pseudocode, since the last two steps (generate blockly and dispatch) are the same